### ISSUE TITLE
SetChargingProfile.req construction fails

### ISSUE THREAD

**Issue by edgale** [mobilityhouse/ocpp#53]
Hi there,

thanks for your work. So far this library works great for me. I set up the following code to limit my charging station to 8A:
```python
#!/usr/bin/python
# -*- coding: utf-8 -*-

import asyncio
from datetime import datetime

try:
    import websockets
except ModuleNotFoundError:
    print("This example relies on the 'websockets' package.")
    print("Please install it by running: ")
    print()
    print(" $ pip install websockets")
    import sys
    sys.exit(1)

from ocpp.routing import on
from ocpp.v16 import ChargePoint as cp
from ocpp.v16.enums import Action, RegistrationStatus
from ocpp.v16 import call_result, call
import ocpp.v16.enums as enums
import ocpp.messages as messages


class ChargePoint(cp):
    @on(Action.BootNotification)
    def on_boot_notitication(self, charge_point_vendor, charge_point_model, **kwargs):
        print('BootNotification')
        return call_result.BootNotificationPayload(
            current_time=datetime.utcnow().isoformat(),
            interval=10,
            status=RegistrationStatus.accepted
        )

    @on(Action.StatusNotification)
    def on_status_notification(self, **kwargs):
        print('status notification')
        return call_result.StatusNotificationPayload()

    @on(Action.Heartbeat)
    def on_heartbeat(self, **kwargs):
        print('heartbeat')
        return call_result.HeartbeatPayload(
            current_time=datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S') + "Z"
        )

    async def send_limitation(self, limit):
        response = await self.call(call.SetChargingProfilePayload(
            connector_id=0,
            cs_charging_profiles={
                'chargingProfileId': 1,
                'stackLevel': 0,
                'chargingProfilePurpose': enums.ChargingProfilePurposeType.chargepointmaxprofile,
                'chargingProfileKind': enums.ChargingProfileKindType.absolute,
                'chargingSchedule': {
                    'startSchedule': datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S') + "Z",
                    'chargingRateUnit': enums.ChargingRateUnitType.amps,
                    'chargingSchedulePeriod': [{
                        'startPeriod': 0,
                        'limit': limit
                    }]
                }
            }
        ))

        print(response.status)


async def on_connect(websocket, path):
    """ For every new charge point that connects, create a ChargePoint instance
    and start listening for messages.

    """
    charge_point_id = path.strip('/')
    cp = ChargePoint(charge_point_id, websocket)

    await asyncio.gather(cp.start(), cp.send_limitation(8.0))


async def main(ip, port):
    server = await websockets.serve(
        on_connect,
        ip,
        port,
        subprotocols=['ocpp1.6']
    )

    await server.wait_closed()


if __name__ == '__main__':
    try:
        # asyncio.run() is used when running this example with Python 3.7 and
        # higher.
        asyncio.run(main('192.168.0.22', 9000))
    except AttributeError:
        # For Python 3.6 a bit more code is required to run the main() task on
        # an event loop.
        loop = asyncio.get_event_loop()
        loop.run_until_complete(main('192.168.0.22', 9000))
        loop.close()
```

Without ```send_limitiation()``` everything works as expected.
After adding ```send_limitiation()``` the charging station receives it. But it seems like it is not correct. Here is what that chargingStation receives (The charging station provides a web interface with a ```last message received``` field):
```json
[2, "9bb54222-ab26-4551-94a1-03b2576885c1", "SetChargingProfile", {
"connectorId": 0,
"csChargingProfiles": {
"chargingProfileId": 1,
"stackLevel": 0,
"chargingProfilePurpose": "ChargePointMaxProfile",
"chargingProfileKind": "Absolute",
"chargingSchedule": {
    "startSchedule": "2019-11-27T13:38:01Z",
    "chargingRateUnit": "A",
    "chargingSchedulePeriod": [{
        "startPeriod": 0,
        "limit": 8.0
    }]
}
}}]: 0, "limit": 8.0}]}}}]
```
The response from the charging station for this message is:
```json
[3,"9bb54222-ab26-4551-94a1-03b2576885c1",{"status":"Accepted"}]
```
Did I do something wrong? Thanks for your help!

**Comment by OrangeTux** [comment]
Hello @edgale. If I understand correctly the charging station seem to receives a request with appended `0, "limit": 8.0}]}}}]`. Am I correct?

I tried your code together with a client I wrote and I am unable to reproduce it. The client code is shown below. Note that I enabled log level DEBUG. This will print the messages send and received over the websocket.  I had the following line in my output:

``` 
DEBUG:websockets.protocol:client < Frame(fin=True, opcode=1, data=b'[2, "00b6c018-ab8b-4ad2-b8c2-a30f27d57d43", "SetChargingProfile", {"connectorId": 0, "csChargingProfiles": {"chargingProfileId": 1, "stackLevel": 0, "chargingProfilePurpose": "ChargePointMaxProfile", "chargingProfileKind": "Absolute", "chargingSchedule": {"startSchedule": "2019-11-27T14:09:56Z", "chargingRateUnit": "A", "chargingSchedulePeriod": [{"startPeriod": 0, "limit": 8.0}]}}}]', rsv1=False, rsv2=False, rsv3=False)
INFO:ocpp:CP_1: receive message [2, "00b6c018-ab8b-4ad2-b8c2-a30f27d57d43", "SetChargingProfile", {"connectorId": 0, "csChargingProfiles": {"chargingProfileId": 1, "stackLevel": 0, "chargingProfilePurpose": "ChargePointMaxProfile", "chargingProfileKind": "Absolute", "chargingSchedule": {"startSchedule": "2019-11-27T14:09:56Z", "chargingRateUnit": "A", "chargingSchedulePeriod": [{"startPeriod": 0, "limit": 8.0}]}}}]
```

My guess is that your charging stations receives the message correctly, but somehow fails to print it correctly. It even response correctly to the message. That would be weird for a charging station to respond correctly on a request which is totally invalid, don't you think?

Here the code for my client:

``` python
import asyncio

try:
    import websockets
except ModuleNotFoundError:
    print("This example relies on the 'websockets' package.")
    print("Please install it by running: ")
    print()
    print(" $ pip install websockets")
    import sys
    sys.exit(1)


from ocpp.v16 import call, call_result
from ocpp.v16 import ChargePoint as cp
from ocpp.routing import on
from ocpp.v16.enums import RegistrationStatus


import logging
logging.basicConfig(level=logging.DEBUG)

class ChargePoint(cp):
    async def send_boot_notification(self):
        request = call.BootNotificationPayload(
            charge_point_model="Optimus",
            charge_point_vendor="The Mobility House"
        )

        response = await self.call(request)

        if response.status ==  RegistrationStatus.accepted:
            print("Connected to central system.")


    @on("SetChargingProfile")
    async def on_set_charging_profile(self, *args, **kwargs):
        return call_result.SetChargingProfilePayload(status="Accepted")


async def main():
    async with websockets.connect(
        'ws://localhost:9000/CP_1',
         subprotocols=['ocpp1.6']
    ) as ws:

        cp = ChargePoint('CP_1', ws)

        await asyncio.gather(cp.start(), cp.send_boot_notification())


if __name__ == '__main__':
    try:
        # asyncio.run() is used when running this example with Python 3.7 and
        # higher.
        asyncio.run(main())
    except AttributeError:
        # For Python 3.6 a bit more code is required to run the main() task on
        # an event loop.
        loop = asyncio.get_event_loop()
        loop.run_until_complete(main())
        loop.close()
```

**Comment by edgale** [comment]
Hi @OrangeTux ,
thank you for your fast reply.
Yes you are correct. The charging station shows an additional ```0, "limit": 8.0}]}}}]```.
I tried your client and enabled debuging in my central system software. Seems like you are right! The message generated and sent correctly to the charging station.
I guess, like you suggested, the charging station just shows it incorrectly. I was worried I did something wrong in my code. But I guess the charging stations just displays something weird.
Thank you for your help and time.
